packages/hash/frontend/src/components/block-loader/block-loader.tsx

```diff
@@ -1,44 +1,41 @@
-import { UnknownRecord } from "@blockprotocol/core";
 import {
-  BlockGraph,
   BlockGraphProperties,
-  EntityType as BpEntityType,
+  EntityEditionId,
+  GraphResolveDepths,
+  Subgraph,
+  SubgraphRootTypes,
 } from "@blockprotocol/graph";
+import { VersionedUri } from "@blockprotocol/type-system/slim";
 import { HashBlockMeta } from "@hashintel/hash-shared/blocks";
-import { JsonSchema } from "@hashintel/hash-shared/json-utils";
-import { EntityId } from "@hashintel/hash-subgraph";
-import { uniqBy } from "lodash";
+import { EntityId } from "@hashintel/hash-shared/types";
+import { Subgraph as LocalSubgraph } from "@hashintel/hash-subgraph";
+import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
 import {
   FunctionComponent,
   useCallback,
+  useEffect,
   useLayoutEffect,
   useMemo,
   useRef,
+  useState,
 } from "react";
-import { useLocalstorageState } from "rooks";
 
 import { useBlockLoadedContext } from "../../blocks/on-block-loaded";
-import { useBlockContext } from "../../blocks/page/block-context";
-import { convertApiEntityToBpEntity } from "../../lib/entities";
 import { useIsReadonlyMode } from "../../shared/readonly-mode";
 import { useBlockProtocolAggregateEntities } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-aggregate-entities";
+import { useBlockProtocolGetEntity } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-get-entity";
+import { useBlockProtocolUpdateEntity } from "../hooks/block-protocol-functions/knowledge/use-block-protocol-update-entity";
 import { useBlockProtocolFileUpload } from "../hooks/block-protocol-functions/use-block-protocol-file-upload";
 import { RemoteBlock } from "../remote-block/remote-block";
-import { DataMapEditor } from "./data-map-editor";
 import { fetchEmbedCode } from "./fetch-embed-code";
-import { SchemaMap } from "./shared";
 
-// @todo consolidate these properties, e.g. take all entityX, linkX into a single childEntity prop
-// @see https://app.asana.com/0/1200211978612931/1202807842439190/f
 type BlockLoaderProps = {
-  blockEntityId: string;
+  blockEntityId?: EntityId; // @todo make this always defined
```

Comment:
Out of curiosity, why did this have to become `undefined`?

If it takes more than a minute or two to explain, don't worry about it.

Reply:
In practice this has always been potentially null. We previously had this prop as required and asserted that it was always defined when calling the component, but then handled it being missing inside this component anyway (current code in `main` [here](https://github.com/hashintel/hash/blob/ae4949d654012828d319816ef8739457c79e97ef/packages/hash/frontend/src/components/block-loader/block-loader.tsx#L83))

This just makes the type reflect reality

Reply:
Gotcha, not ideal